include/seqan3/range/container/bitcompressed_vector.hpp

```diff
@@ -0,0 +1,1101 @@
+     *
+     * This element acts as a placeholder; attempting to dereference it results in undefined behaviour.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    iterator end() noexcept
+    {
+        return iterator{*this, size()};
+    }
+
+    //!\copydoc end()
+    const_iterator end() const noexcept
+    {
+        return const_iterator{*this, size()};
+    }
+
+    //!\copydoc end()
+    const_iterator cend() const noexcept
+    {
+        return const_iterator{*this, size()};
+    }
+    //!\}
+
+    /*!\name Element access
+     * \{
+     */
+    /*!\brief Return the i-th element.
+     * \param[in]              i Index of the element to retrieve.
+     * \throws std::out_of_range If you access an element behind the last.
+     * \returns                  Either a writable proxy to the element or a copy (if called in const context).
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * Strong exception guarantee (never modifies data).
+     */
+    reference at(size_type const i)
+    {
+        if (i >= size()) // [[unlikely]]
+        {
+            throw std::out_of_range{"Trying to access element behind the last in bitcompressed_vector."};
+        }
+        return (*this)[i];
+    }
+
+    //!\copydoc at()
+    const_reference at(size_type const i) const
+    {
+        if (i >= size()) // [[unlikely]]
+        {
+            throw std::out_of_range{"Trying to access element behind the last in bitcompressed_vector."};
+        }
+        return (*this)[i];
+    }
+
+    /*!\brief Return the i-th element.
+     * \param i The element to retrieve.
+     * \returns Either a writable proxy to the element or a copy (if called in const context).
+     *
+     * Accessing an element behind the last causes undefined behaviour. In debug mode an assertion checks the size of
+     * the container.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * Strong exception guarantee (never modifies data)..
+     */
+    reference operator[](size_type const i) noexcept(noexcept(data[i]))
+    {
+        assert(i < size());
+        return data[i];
+    }
+
+    //!\copydoc operator[]()
+    const_reference operator[](size_type const i) const noexcept(noexcept(assign_rank(const_reference{}, data[i])))
+    {
+        assert(i < size());
+        return assign_rank(const_reference{}, data[i]);
+    }
+
+    /*!\brief Return the first element. Calling front on an empty container is undefined.
+     * \returns Either a writable proxy to the element or a copy (if called in const context).
+     *
+     * Calling front on an empty container is undefined. In debug mode an assertion checks the size of the container.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * Strong exception guarantee (never modifies data).
```

Comment:
Actually the documentation is a little bit tricky, as either strong_exeception guarantee is given or never throws depending on the evaluation of the expression. I think the STL also writes this in a similar way.